fmt_cucumber.go

```diff
@@ -122,115 +121,86 @@ type cukefmt struct {
 	// of the example name inorder to build id fields.
 }
 
-func (f *cukefmt) Node(n interface{}) {
-	f.basefmt.Node(n)
+func (f *cukefmt) Pickle(pickle *messages.Pickle) {
+	f.basefmt.Pickle(pickle)
 
-	switch t := n.(type) {
+	scenario := f.findScenario(pickle.AstNodeIds[0])
```

Comment:
Should there be a `if scenario == nil` check here? Otherwise wouldn't line 133 panic?

Reply:
I'm not sure, if `f.findScenario` doesn't find a scenario, something is really broken, but we could add something like this:
```go
if scenario == nil {
  panic("couldn't find scenario for AST ID: " + astNodeID)
}
```

to improve debugging

Reply:
what do you think about this:
```
func (f *basefmt) findScenario(scenarioAstID string) *messages.GherkinDocument_Feature_Scenario {
	for _, ft := range f.features {
		if sc := ft.findScenario(scenarioAstID); sc != nil {
			return sc
		}
	}

	panic("couldn't find scenario for AST ID: " + scenarioAstID)
}
```

Reply:
👍